packages/api/cms-api/src/dam/files/file-upload.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from "@nestjs/common";
+import FileType from "file-type";
+import fs from "fs";
+import got from "got";
+import os from "os";
+import stream from "stream";
+import { promisify } from "util";
+import { v4 as uuid } from "uuid";
+
+import { FileUploadInterface } from "./dto/file-upload.interface";
+import { FilesService } from "./files.service";
+
+const pipeline = promisify(stream.pipeline);
+
+@Injectable()
+class FileUploadService {
+    async createUploadFromUrl(url: string): Promise<FileUploadInterface> {
```

Comment:
Why `createUploadFromUrl` and not just `uploadFromUrl`?

Reply:
> (To me createUploadFromUrl implies that the upload is now created but I still have to do a second step to execute the upload and get the actual file)

That's exactly what you have to do, as the actual upload happens in `FilesService#upload`

Reply:
Oh, I misread your concern, sorry. `uploadFromUrl` sounds like "upload this now" to me. It's the same for `createUploadFromUrl`. Maybe find something new?

Reply:
Suggestions:

1. createUploadFromUrl
2. uploadFromUrl
3. createFileToUploadFromUrl
4. createUploadableFromUrl

@nsams @thomasdax98 what do you prefer?

Reply:
createUploadInputFromUrl?

Reply:
@thomasdax98 what do you think about `createUploadInputFromUrl?`?

Maybe we should rename FileUploadInterface as well? Why isn't this called `FileUploadInput`? 🤔

Reply:
Why not rename it to `FileUploadInput` and the method to `createFileUploadInputFromUrl`. That would be coherent in my opinion

Reply:
Done: 6129691c10202fab55673e61eb42a21c46c0ac6c